src/async_client.rs

```diff
@@ -75,6 +80,33 @@ impl AsyncClient {
         Ok(client.state())
     }
 
+    /// Disconnects from endpoint.
+    ///
+    /// This consumes the client and handles the graceful shutdown of the connection. This should be
+    /// preferred over simply dropping the instance to give the server a chance to clean up and also
+    /// to avoid blocking unexpectedly when the client is being dropped without calling this method.
+    pub async fn disconnect(mut self) {
+        log::info!("Disconnecting from endpoint");
+
+        let status_code = ua::StatusCode::new({
+            let Ok(mut client) = self.client.lock() else {
+                log::warn!("Error while disconnecting client");
+                return;
+            };
+
+            unsafe { UA_Client_disconnectAsync(client.as_mut_ptr()) }
+        });
+        if let Err(error) = Error::verify_good(&status_code) {
+            log::warn!("Error while disconnecting client: {error}");
+        }
+
+        // Wait until background task has finished. The `take()` should always succeed because there
+        // is no other location where the handle is being consumed.
+        if let Some(background_handle) = self.background_handle.take() {
+            let _unused = background_handle.await;
```

Comment:
- Is it guaranteed that this always terminated?
- What happens if disconnecting failed?
- Should we explicitly `abort()` the background task like `drop()` does?